### PR TITLE
Telemetry event

### LINKED ISSUE #16417: Ensure all telemetry events have the same metadata
## Expected Behavior
Ensure that all Posthog events contain the necessary metadata, in this case the node leader version, so we can reliably tell deployments' upgrade lag.

## Current Behavior
When looking at upgrade behavior, I noticed that `Identify` and `Group Identify` events don't contain the leader node's version, making it hard to understand the development of deployed versions over time.

## Possible Solution
Ideally, deliver all static metadata in the HTML page directly, so this metadata is available to all, even very early, telemetry events.

## Steps to Reproduce (for bugs)
1. Check a `Identify` event and look for the flattened semver fields (`node_leader_app_version__build` for example)

## Your Environment

* Graylog Version: master

### PR DESCRIPTION
**Pull request by ousmaneo**
This PR introduces custom-named events for telemetry, In the previous version, we used generic event names, which was not easy to use in Posthog. 

fix #16417 
/jpd Graylog2/graylog-plugin-enterprise#5691
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by grotlue — CHANGES_REQUESTED** [review]
In general it looks good. While testing I found some issues. I think a few just be fixed (the bug for example) but some would be nice to have and we can also discuss/create follow-up issues for that

## Test results
I tested some general tracking test cases. Additionally, I picked sections based on the code and did regression testing on those due to time constraints.
Let me know if I should test something specific or anything else.

### General
✅ Pageview is tracked as usual
✅ Group Identify is tracked as usual
✅ Tracking is not fired when disabled
✅ Opt out event is tracked
✅ Opt in event is tracked

### Streams
❌ Click on start/stop stream in the status column is not tracked
❌ `Stream Item Clone Modal Opened` is tracked twice
❌ `Stream Item Deleted` is tracked when clicking on Delete even when clicking cancel and not actually deleting the stream afterwards

### Search
❌ multiple actions are still tracked with generic events (`input_button_toggle`, `click`, `form_submit`)

### Rules & Pipelines
No issues found
### Events
I wasn't able to test tracking when events and notifications are created due to the bug

❌ When attempting to create a simple event definition with a `value=true` search filter I encountered the following error: 
Screenshot 2023-09-28 at 12 46 16

❌ No tracking for create event notification button click event (but we have it for (`EventDefinition Create Button Clicked`)

### Content Stream
❌ Opening a news article and a release are both tracked with `Content Stream Article Clicked` other clicks are categorized into Releases and News. It would be good to track a release article click separately. 

### Dashboards
No issues found

**Comment by ousmaneo** [comment]
Hi Thank you for reviewing,

> ❌ multiple actions are still tracked with generic events (input_button_toggle, click, form_submit)

Do you have some example for `input_button_toggle` and `click` I only found 2 `form_submit` event?

**Comment by grotlue** [comment]
> Hi @grotlue Thank you for reviewing,
> 
> > ❌ multiple actions are still tracked with generic events (input_button_toggle, click, form_submit)
> 
> Do you have some example for `input_button_toggle` and `click` I only found 2 `form_submit` event?

Oh I should have written them down. One example is search filter create for `click`.

**Comment by ousmaneo** [comment]
@grotlue I think you were missing the enterprise branch. It was linked from the enterprise branch but not on the dependency list on this one.

File: graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config [hunk 1]
```diff
@@ -69,7 +69,7 @@ cloud_inputs=on
 preflight_web=off
 
 # Instant archiving is disabled by default
-instant_archiving=off
+instant_archiving=on
```
[the remark this answers is not in the document]
Comment (ousmaneo) [reply graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config]: I reverted it after noticing

File: graylog2-web-interface/src/logic/telemetry/TelemetryProvider.tsx [hunk 2]
```diff
@@ -90,9 +90,10 @@ const TelemetryProvider = ({ children }: { children: React.ReactElement }) => {
           ...license,
           ...plugin,
           ...searchCluster,
+          ...getGlobalProps(telemetryData),
         });
 
-        posthog.identify(user, { ...userDetails });
+        posthog.identify(user, { ...userDetails, ...getGlobalProps(telemetryData) });
```
[the remark this answers is not in the document]
Comment (ousmaneo) [reply graylog2-web-interface/src/logic/telemetry/TelemetryProvider.tsx:96]: These contain different data so I think it is safe.

File: graylog2-web-interface/src/components/event-definitions/event-definition-form/EventDefinitionFormContainer.tsx [hunk 3]
```diff
@@ -138,15 +144,33 @@ const EventDefinitionFormContainer = ({ action, eventDefinition: eventDefinition
 
   const handleSubmit = () => {
     if (action === 'create') {
+      sendTelemetry(TELEMETRY_EVENT_TYPE.EVENTDEFINITION_SUMMARY.CREATE_CLICKED, {
+        app_pathname: getPathnameWithoutId(pathname),
+        app_section: 'new-event-definition',
```
[the remark this answers is not in the document]
Comment (ousmaneo) [reply graylog2-web-interface/src/components/event-definitions/event-definition-form/EventDefinitionFormContainer.tsx:149]: For the scope of this PR, we decided to change only the event names and some `event_details`. We might update how we set the app_section in the follow-up work.

**Comment by grotlue** [comment]
> @grotlue I think you were missing the enterprise branch. It was linked from the enterprise branch but not on the dependency list on this one.

Ah yes I did not know about that one.

File: graylog2-web-interface/src/components/streams/StreamsOverview/cells/StatusCell.tsx [hunk 4]
```diff
@@ -51,8 +53,17 @@ const StatusCell = ({ stream }: Props) => {
   const disableChange = stream.is_default || !stream.is_editable;
   const description = stream.disabled ? 'Paused' : 'Running';
   const title = _title(stream.disabled, disableChange, description);
+  const sendTelemetry = useSendTelemetry();
 
   const toggleStreamStatus = useCallback(async () => {
+    sendTelemetry(TELEMETRY_EVENT_TYPE.STREAMS.STREAM_ITEM_STATUS_TOGGLED, {
+      app_pathname: 'streams',
+      app_action_value: ' stream-item-status',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/streams/StreamsOverview/cells/StatusCell.tsx:61]:
```suggestion
      app_action_value: 'stream-item-status',
```

**Comment by grotlue** [comment]
Thanks for the fixes, I already tested the following issues (checked == fixed) and they look good now!
However, one new issue came up:

❌ Stream deletion is not tracked at all anymore (also when actually deleting the stream)


### Streams

- [x] ❌ Click on start/stop stream in the status column is not tracked
- [x] ❌ Stream Item Clone Modal Opened is tracked twice
- [x] ❌ Stream Item Deleted is tracked when clicking on Delete even when clicking cancel and not actually deleting the stream afterwards


### Search

- [x] ❌ multiple actions are still tracked with generic events (input_button_toggle, click, form_submit)


### Content Stream

- [x] ❌ Opening a news article and a release are both tracked with Content Stream Article Clicked other clicks are categorized into Releases and News. It would be good to track a release article click separately.

**Review by grotlue — APPROVED** [review]
Had another look, thanks for fixing all those issues! 👍 

## The following are fixed:
> ❌ When attempting to create a simple event definition with a value=true search filter I encountered the following error:
> ❌ Stream deletion is not tracked at all anymore (also when actually deleting the stream)


## Open issues
The tracking for event creation also looks good. The only thing left here is the tracking for event notification creation. 
I think we can still merge all the other things and fix it in another PR, but we should create an issue for that.

**Comment by linuspahl** [comment]
@ousmaneo @grotlue I saw there are some cases where we are now using a `button` instead of an `anchor` for links. For example for the "Create event definition" link. We mainly removed the `LinkContainer`. Couldn't we just add `onClick` for the `Button` or `LinkContainer` component?

My concern is that this change has some disadvantages, like worse accessibility. For example I noticed you can no longer do "open in new tab" by right clicking on the link and selecting the option or by pressing Command or Ctrl and clicking on the link.

**Comment by ousmaneo** [comment]
> @ousmaneo @grotlue I saw there are some cases where we are now using a `button` instead of an `anchor` for links. For example for the "Create event definition" link. We mainly removed the `LinkContainer`. Couldn't we just add `onClick` for the `Button` or `LinkContainer` component?
> 
> My concern is that this change has some disadvantages, like worse accessibility. For example I noticed you can no longer do "open in new tab" by right clicking on the link and selecting the option or by pressing Command or Ctrl and clicking on the link.


Hi @linuspahl, Thank you for your comment. I think you are right instead of changing the component of the` Create event definition` to a button, we could add a onClick to the child button to trigger the telemetry call. The LinkContainer is already designed to call the onClick on the child button. I will create an Issue for it.

**Pull request merged**